src/Terrain/terrain.cpp

```diff
@@ -164,8 +164,8 @@ void Terrain::init_area(int area_x, int area_y, LandGenerator gen){
 
 void Terrain::init_chunks(){
     for (int xyz=0; xyz<((X_MAX-1)/Chunk::size+1)*((Y_MAX-1)/Chunk::size+1)*((Z_MAX-1)/Chunk::size+1); xyz+=1){
-        auto [x,y,z] = sop(xyz, ((X_MAX-1)/Chunk::size+1),((Y_MAX-1)/Chunk::size+1),((Z_MAX-1)/Chunk::size+1));
-        chunks.push_back(Chunk(x,y,z, this));
+        auto xyz_sop = sop(xyz, ((X_MAX-1)/Chunk::size+1),((Y_MAX-1)/Chunk::size+1),((Z_MAX-1)/Chunk::size+1));
+        chunks.push_back(Chunk(xyz_sop[0],xyz_sop[1],xyz_sop[2], this));
```

Comment:
What's wrong with destructuring? Did you set your C++ version?

https://en.cppreference.com/w/cpp/language/structured_binding